### PR TITLE
Ss/jig 232 refactor authless client

### PR DESCRIPTION
**Pull request by 0ff1ane**
## Proposed changes

Removes caching behaviour from authless-client.

## Types of changes

What types of changes does your code introduce?

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

None.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -1,118 +1,87 @@
...
 
-  cache: Cache | null
+  uri: string
   retries: number
-  lowLevelClient: ClientLowLevel
+  axios: AxiosStatic
 
-  static isValidConfig (x: any): x is Config {
-    if(!(x instanceof Object)) throw new Error('config must be a javascript object')
+  static isValidClient (x: any): x is Client {
```
Comment (MichaelHirn) [inline src/client.ts]: shouldn't this be 

```suggestion
  static isValidClient (x: any): x is Config {
```

File: src/client.ts [hunk 2]
```diff
@@ -1,118 +1,87 @@
...
+
+const debug = Debug.extend('client')
 
 interface Config {
-  cache?: Cache
-  retries: number
-  lowLevelClient: ClientLowLevel
+  uri: any
+  retries: any
 }
```
Comment (MichaelHirn) [inline src/client.ts]: would be good if it also allows optionally to provide axios instance - might be helpful for providing mocks or a pre-configured axios instance.

File: src/client.ts [hunk 3]
```diff
@@ -1,118 +1,87 @@
...
   }
+
+  async search (params: SearchParams): Promise<ResponseData> {
+    debug.extend('search')(params.q)
+    params.responseFormat = params.responseFormat ?? 'json'
+
+    return await this.axios.get(`${this.uri}/search`, {params})
+      .then(response => response.data)
+  }
 }
```
Comment (MichaelHirn) [inline src/client.ts]: Is `/search` a valid endpoint at the authless-server anymore? I don't think this is supported anymore, so we could remove this here.

File: src/client.ts [hunk 4]
```diff
@@ -1,118 +1,87 @@
...
 import Debug from './debug'
-const debug = Debug.extend('high')
+
+const debug = Debug.extend('client')
 
 interface Config {
-  cache?: Cache
-  retries: number
-  lowLevelClient: ClientLowLevel
+  uri: any
```
Comment (MichaelHirn) [inline src/client.ts]: Would be good to fina a more descriptive name for this, e.g. `serverEndpoint`, `serverUrl`, `serverAddress`, ...

**Review by MichaelHirn — CHANGES_REQUESTED** [review]
Once you addressed the 4 issues it should be ready to get merged.

File: src/client.ts (hunk 3 above)
Comment (0ff1ane) [reply src/client.ts]: done

File: src/client.ts (hunk 4 above)
Comment (0ff1ane) [reply src/client.ts]: made it `serverUri`

**Review by MichaelHirn — APPROVED** [review]

**Comment by MichaelHirn** [comment]
closing this PR in favor of #7, which has a cleaned git history (rebased properly onto origin/master & and squashed commits)

**Pull request closed without merging**